iceoryx_utils/test/moduletests/test_cxx_algorithm.cpp

```diff
@@ -66,3 +71,134 @@ TEST_F(algorithm_test, MinOfManyElements)
+    {
+        a.emplace_back(i);
+    }
+
+    for (int64_t i = 5; i < 10; ++i)
+    {
+        b.emplace_back(i);
+    }
+
+    auto mergedContainer = uniqueMergeSortedContainers(a, b);
+
+    ASSERT_THAT(mergedContainer.size(), Eq(10U));
+    for (int64_t i = 0; i < 10; ++i)
+        EXPECT_THAT(mergedContainer[i], Eq(i));
+}
+
+TEST_F(algorithm_test, MergeTwoAlternatingDisjunctNonEmptySortedContainers)
+{
+    vector<int64_t, 10U> a, b;
+
+    for (int64_t i = 0; i < 5; ++i)
+    {
+        a.emplace_back(i * 2);
+    }
+
+    for (int64_t i = 0; i < 5; ++i)
+    {
+        b.emplace_back(i * 2 + 1);
+    }
+
+    auto mergedContainer = uniqueMergeSortedContainers(a, b);
+
+    ASSERT_THAT(mergedContainer.size(), Eq(10U));
+    for (int64_t i = 0; i < 10; ++i)
+        EXPECT_THAT(mergedContainer[i], Eq(i));
+}
+
+TEST_F(algorithm_test, MergingIdenticalContainerResultsInUnchangedContainer)
+{
+    vector<int64_t, 10U> a;
+
+    for (int64_t i = 0; i < 5; ++i)
+    {
+        a.emplace_back(i * 2);
+    }
+
+    auto mergedContainer = uniqueMergeSortedContainers(a, a);
+
+    ASSERT_THAT(mergedContainer.size(), Eq(5U));
+    for (int64_t i = 0; i < 5; ++i)
+        EXPECT_THAT(mergedContainer[i], Eq(i * 2));
+}
+
+TEST_F(algorithm_test, MergingWithOneEmptyContainerResultsInUnchangedContainer)
+{
+    vector<int64_t, 10U> a;
+
+    for (int64_t i = 0; i < 5; ++i)
+    {
+        a.emplace_back(i * 3);
+    }
+
+    auto mergedContainer = uniqueMergeSortedContainers(a, vector<int64_t, 10U>());
+
+    ASSERT_THAT(mergedContainer.size(), Eq(5U));
+    for (int64_t i = 0; i < 5; ++i)
+        EXPECT_THAT(mergedContainer[i], Eq(i * 3));
+}
+
+TEST_F(algorithm_test, MergePartiallyOverlappingSortedContainers)
+{
+    vector<int64_t, 10U> a, b;
+
+    for (int64_t i = 3; i < 10; ++i)
+    {
+        a.emplace_back(i);
+    }
+
+    for (int64_t i = 0; i < 8; ++i)
+    {
+        b.emplace_back(i);
+    }
+
+    auto mergedContainer = uniqueMergeSortedContainers(a, b);
+
+    ASSERT_THAT(mergedContainer.size(), Eq(10U));
+    for (int64_t i = 0; i < 10; ++i)
+        EXPECT_THAT(mergedContainer[i], Eq(i));
+}
+
+TEST_F(algorithm_test, MergeWithDisjunctOneElementContainer)
+{
+    vector<int64_t, 10U> a, b;
+
+    for (int64_t i = 0; i < 5; ++i)
+    {
+        a.emplace_back(i);
+    }
+
+    b.emplace_back(5);
+
+    auto mergedContainer = uniqueMergeSortedContainers(a, b);
+
+    ASSERT_THAT(mergedContainer.size(), Eq(6U));
+    for (int64_t i = 0; i < 6; ++i)
+        EXPECT_THAT(mergedContainer[i], Eq(i));
+}
+
+TEST_F(algorithm_test, MergeWithOverlappingOneElementContainer)
+{
+    vector<int64_t, 10U> a, b;
+
+    for (int64_t i = 0; i < 5; ++i)
+    {
+        a.emplace_back(i);
+    }
+
+    b.emplace_back(0);
+
+    auto mergedContainer = uniqueMergeSortedContainers(a, b);
+
+    ASSERT_THAT(mergedContainer.size(), Eq(5U));
+    for (int64_t i = 0; i < 5; ++i)
+        EXPECT_THAT(mergedContainer[i], Eq(i));
+}
```

Comment:
I think a test with a gap in the numbers could be interesting, e.g. `a = [0,1,2];` and `b = [7,8,9];`.
Additionally, some of the tests should also check with switched input like `uniqueMergeSortedContainers(b, a);`